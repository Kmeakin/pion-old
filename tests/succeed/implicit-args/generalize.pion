//~ run = "${PION} elab ${FILE}"

{
let id : fun (@A : Type) -> A -> A = fun a => a;
let always : fun (@A : Type) (@B : Type) -> A -> B -> A = fun a b => a;
let apply : fun (@A : Type) (@B : Type) -> (A -> B) -> A -> B = fun f x => f(x);
{}
}
