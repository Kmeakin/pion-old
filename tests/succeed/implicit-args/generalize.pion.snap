exit status: 0
stdout:
{
    let id : fun (@A : Type) -> A -> A = fun (@A : Type) (a : A) => a;
    let always : fun (@A : Type) (@B : Type) -> A -> B -> A = fun (@A : Type)
    (@B : Type)
    (a : A)
    (b : B) => a;
    let apply : fun (@A : Type) (@B : Type) -> (A -> B) -> A ->
    B = fun (@A : Type) (@B : Type) (f : A -> B) (x : A) => f x;
    ()
} : ()

stderr:
